pipewire: 0.3.42 -> 0.3.43

jansol opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

https://gitlab.freedesktop.org/pipewire/pipewire/-/releases/0.3.43

- pulse-server should use less memory and overall work better
- filter-chain now supports lv2 effects
- libcamera bump since it had an API change that affects spa-libcamera
- Works On My Machine (tm)

NB: backport requires #153124 to land first (feel free to retarget it directly to the release branch, I may have been overly cautious with sending it to staging)

https://git.libcamera.org/libcamera/libcamera.git/log/?id=40f5fddca7f774944a53f58eeaebc4db79c373d8..1db1e31e664c1f613dc964d8519fe75d67b154b6

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

jansol committed:
libcamera: 2021-09-24 -> 2022-01-03

jansol committed:
pipewire: 0.3.42 -> 0.3.43

K900 commented:
LGTM and is needed for latest Wireplumber (see linked PR)

jtojnar commented on pkgs/development/libraries/libcamera/default.nix:
```diff
@@ -65,7 +81,11 @@ stdenv.mkDerivation {
 
-  mesonFlags = [ "-Dv4l2=true" "-Dqcam=disabled" ];
+  mesonFlags = [
+    "-Dv4l2=${mesonBool true}"
```
Why add functions when the values are just static?

jansol replied:
Future proofing mostly (for making the features accessible as parameters to the derivation), and to make the types of the options more explicit.

jtojnar replied:
I would not say it does a very good job in either – it adds clutter to the expression that might (or might not) be used in the future and actually obscures the value that is passed to the options. And the type that accepts `enabled`/`disabled` is called `feature`.

jansol replied:
Fair enough, I'll just remove it for now. And rename the one in the pipewire expression (where I copied this from) to mesonFeature.

jtojnar commented on pkgs/development/libraries/libcamera/default.nix:
```diff
@@ -44,13 +54,19 @@ stdenv.mkDerivation {
     gtest
   ];
 
+  FONTCONFIG_FILE = fontsConf;
```
Probably should go next to the other env vars. And a comment why it is needed would be useful.

jansol replied:
I'm not even sure if this is the proper way to get rid of the fontconfig errors, it's just what I found was being done in other packages. It only gets rid of the "no config file" errors anyway, there are still like a thousand lines of "no writable cache".

jtojnar replied:
Yeah, this is the proper “fix”. I am mostly interested what needs fonts in libcamera. It is typically needed by GUI tests or sometimes docs generation.

jansol replied:
Probably docs yes, unless gstreamer requires it.

jtojnar replied:
What commands precede the warning?

jansol replied:
Seems indeed to be to the documentation. Build log says
```
[256/378] Generating doxygen with a custom commandap_ipa_proxyxyrialization_test.cpp.oK[Kerializer_test.cpp.o.o
/build/libcamera-1db1e31/src/libcamera/base/file.cpp:49: warning: no matching class member found for
  libcamera::File::MapFlag::NoOption
<a few more similar warnings but this looks like it's interspersed here from another command running in parallel>
sh: latex: not found
error: Problems running latex. Check your installation or look for typos in _formulas.tex and check _formulas.log!
sh: dvips: not found
error: Problems running dvips. Check your installation!
Fontconfig error: No writable cache directories
Fontconfig error: No writable cache directories
Fontconfig error: No writable cache directories
...
```

The documentation is built with sphinx, which does mention LaTeX as one backend (for pdf rendering) right on its landing page. There is no option for the documentation format and the `meson.build` for the documentation specifically requests html output so my best guess is that this is sphinx probing the environment to determine which backends are actually usable.

jansol commented:
Removed the systemd unit path patches that were backported to master the other day that are part of the upstream release.

jansol commented:
Renamed the helper to `mesonEnableFeature` to match the wireplumber PR, inlined the fontconfig workaround, removed `patchShebang` invocations for files that no longer exist.

jansol commented:
@Artturin release-21.11 and nixos-unstable have diverged a bit by now so it's probably easiest to just copy the current state of nixos-unstable into the stable branch (minus the change of defaulting to wireplumber instead of media-session). Remember that libcamera also needs to be updated.
